hydra-node/src/Hydra/Network/BroadcastToSelf.hs

```diff
@@ -0,0 +1,22 @@
+module Hydra.Network.BroadcastToSelf where
+
+import Cardano.Prelude
+import Control.Monad.Class.MonadThrow (MonadThrow)
+import Hydra.Network (Network (..), NetworkCallback, NetworkComponent)
+
+-- | Ensures that messages `broadcast` through the wrapped `NetworkComponent` are
+-- also sent back to "self" through passed `NetworkCallback`.
+withBroadcastToSelf ::
+  MonadThrow m =>
+  NetworkComponent m msg ->
+  NetworkComponent m msg
+withBroadcastToSelf withNetwork callback action =
```

Comment:
So.. I struggle a bit in deciphering the arguments vs the type signature. I know that the `NetworkComponent` is both, a callback and a continuation.. but it is kinda hard to read here. Not sure what to do about it though.

Reply:
Yeah, that's also something I find annoying with aliasing function types: Arguments seem to come out of nowhere. And like you I am not sure what do to about it